perun-rpc/src/main/java/cz/metacentrum/perun/rpc/Api.java

```diff
@@ -243,7 +249,6 @@ private static PerunPrincipal setupPerunPrincipal(HttpServletRequest req, Deseri
 
 		// If OIDC_CLAIM_sub header is present, it means user authenticated via OAuth2 with MITRE.
 		else if (isNotEmpty(req.getHeader(OIDC_CLAIM_SUB))) {
```

Comment:
This is where main logic is happening - this is where a lot of things can go wrong